sources/packages/backend/workflow/test/2023-2024/parttime-assessment/costs/parttime-assessment-2023-2024-additional-transportation-costs.e2e-spec.ts

```diff
@@ -0,0 +1,339 @@
+import { PROGRAM_YEAR } from "../../constants/program-year.constants";
+import {
+  createFakeConsolidatedPartTimeData,
+  executePartTimeAssessmentForProgramYear,
+} from "../../../test-utils";
+import { OfferingDeliveryOptions } from "@sims/test-utils";
+
+describe(`E2E Test Workflow parttime-assessment-${PROGRAM_YEAR}-additional-transportation-costs.`, () => {
+  it(
+    "Should determine calculatedDataTotalTransportationAllowance when studentDataAdditionalTransportKm is zero " +
+      "and offeringDelivered is onsite",
+    async () => {
+      // Arrange
+      const assessmentConsolidatedData =
+        createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+      assessmentConsolidatedData.studentDataAdditionalTransportKm = 0;
+      assessmentConsolidatedData.offeringWeeks = 30;
+      assessmentConsolidatedData.studentDataAdditionalTransportWeeks = 30;
+      assessmentConsolidatedData.studentDataAdditionalTransportCost = 30;
+      assessmentConsolidatedData.offeringDelivered =
+        OfferingDeliveryOptions.Onsite;
+      // Act
+      const calculatedAssessment =
+        await executePartTimeAssessmentForProgramYear(
+          PROGRAM_YEAR,
+          assessmentConsolidatedData,
+        );
+      // Assert
+      // calculatedDataTotalTransportationAllowance = 13 * 30 = 390
+      expect(
+        calculatedAssessment.variables
+          .calculatedDataTotalTransportationAllowance,
+      ).toBe(390);
+    },
+  );
+
+  it(
+    "Should determine calculatedDataTotalTransportationAllowance when studentDataAdditionalTransportKm is zero " +
+      "and offeringDelivered is online",
+    async () => {
+      // Arrange
+      const assessmentConsolidatedData =
+        createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+      assessmentConsolidatedData.studentDataAdditionalTransportKm = 0;
+      assessmentConsolidatedData.offeringWeeks = 30;
+      assessmentConsolidatedData.studentDataAdditionalTransportWeeks = 30;
+      assessmentConsolidatedData.studentDataAdditionalTransportCost = 30;
```

Comment:
IMO, lines 44, 46 and 47 are not required for this test. Likewise, for other program years too!

Reply:
The extra variables are removed in the latest commit. Please have a look.